.github/workflows/browserstack-run.yaml

```diff
@@ -0,0 +1,123 @@
+    outputs:
+      browserstack-build-id:
+        description: The newly created BrowserStack build ID
+        value: ${{ jobs.browserstack-run.outputs.browserstack_build_id }}
+    secrets:
+      BROWSERSTACK_USERNAME:
+        description: BrowserStack Username
+        required: true
+      BROWSERSTACK_ACCESS_KEY:
+        description: BrowserStack Access Key
+        required: true
+      SLACK_WEBHOOK:
+        description: Slack Notifier Incoming Webhook
+        required: true
+
+jobs:
+  browserstack-run:
+    runs-on: ubuntu-latest
+    outputs:
+      browserstack_build_id: ${{ steps.start_run.outputs.browserstack_build_id }}
+
+    steps:
+      # Get the test artifacts prepared in previous step
+      - name: Get the app-debug-signed.apk artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: app-debug-signed.apk
+
+      - name: Get the davinci-debug-androidTest-signed.apk artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: davinci-debug-androidTest-signed.apk
+
+      - name: Unzip artifacts (optional)
+        run: |
+          unzip -o app-debug-signed.apk || echo "Skipping unzip for app-debug"
+          unzip -o davinci-debug-androidTest-signed.apk || echo "Skipping unzip for test apk"
+
+      # Upload app to BrowserStack
+      - name: Upload app-debug-signed.apk to BrowserStack
+        id: upload_app
+        run: |
+          APP_UPLOAD_RESPONSE=$(curl -u "${{ secrets.BROWSERSTACK_USERNAME }}:${{ secrets.BROWSERSTACK_ACCESS_KEY }}" \
+            -X POST "https://api-cloud.browserstack.com/app-automate/espresso/v2/app" \
+            -F "file=@app-debug-signed.apk")
+          echo "APP_URL=$(echo $APP_UPLOAD_RESPONSE | jq -r '.app_url')" >> $GITHUB_ENV
+
+      # Upload test suite APK to BrowserStack
+      - name: Upload davinci-debug-androidTest-signed.apk to BrowserStack
+        id: upload_test_suite
+        run: |
+          TEST_UPLOAD_RESPONSE=$(curl -u "${{ secrets.BROWSERSTACK_USERNAME }}:${{ secrets.BROWSERSTACK_ACCESS_KEY }}" \
+            -X POST "https://api-cloud.browserstack.com/app-automate/espresso/v2/test-suite" \
+            -F "file=@davinci-debug-androidTest-signed.apk")
+          echo "TEST_SUITE_URL=$(echo $TEST_UPLOAD_RESPONSE | jq -r '.test_suite_url')" >> $GITHUB_ENV
+
+      # Prepare BrowserStack run configuration. 
+      # To run tests in parallel on multiple devices, use the "shards" option. Example:
+      #      "shards": {
+      #        "numberOfShards": 3,
+      #        "deviceSelection": "all"
+      #      }
+      #
+      - name: Prepare BrowserStack run configuration
+        run: |
+          DEVICES_JSON=$(echo '${{ vars.BS_DEVICES }}' | jq -R 'split(",")')
+          cat <<EOF > browserstack-run-config.json
+          {
+            "app": "${{ env.APP_URL }}",
+            "testSuite": "${{ env.TEST_SUITE_URL }}",
+            "singleRunnerInvocation": ${{ vars.BS_SINGLE_RUNNER_INVOCATION }},
```

Comment:
`BS_SINGLE_RUNNER_INVOCATION` is repo variable - by default it is set to `true`. 
More details here: https://www.browserstack.com/docs/app-automate/espresso/single-runner-invocation